packages/cmr-client/tests/test-EarthdataLogin.js

```diff
@@ -18,160 +27,306 @@ test.after.always(() => {
 
-test.serial('getToken returns a valid token', async (t) => {
-  const edlObj = new EarthdataLogin({
-    username: 'username',
-    password: 'password',
-    edlEnv: 'PROD',
+test('createEDLToken returns the access token', async (t) => {
+  const { username, password, postResponse } = t.context;
+
+  nock('https://sit.urs.earthdata.nasa.gov')
+    .post('/api/users/token')
+    .reply(200, postResponse);
+
+  const createdToken = await createEDLToken(username, password, 'SIT');
+
+  t.is(createdToken, postResponse.access_token);
+});
+
+test('createEDLToken sends the correct credentials', async (t) => {
+  const { username, password, postResponse } = t.context;
+
+  const scope = nock('https://sit.urs.earthdata.nasa.gov')
+    .post('/api/users/token')
+    .reply(200, postResponse);
+
+  t.plan(1);
+
+  scope.on('request', (req) => {
+    t.is(req.headers.authorization, buildBasicAuthHeader(username, password));
   });
 
-  const expectedresponse = [
-    {
-      access_token: 'ABCDE',
-      token_type: 'Beaer',
-      expiration_date: '1/1/2999',
-    },
-  ];
+  await createEDLToken(username, password, 'SIT');
+});
 
-  nock('https://urs.earthdata.nasa.gov')
+test('retrieveEDLToken returns undefined if the returned token is expired', async (t) => {
+  const { username, password } = t.context;
+
+  const now = new Date();
+  const oneHourAgo = new Date(now.valueOf() - (60 * 60 * 1000));
+
+  const expiredToken = createToken({
+    expirationTime: oneHourAgo.valueOf() / 1000,
+  });
+
+  const expirationDate = oneHourAgo.toLocaleDateString('en', {
+    month: '2-digit',
+    day: '2-digit',
+    year: 'numeric',
+  });
+
+  nock('https://sit.urs.earthdata.nasa.gov')
     .get('/api/users/tokens')
-    .reply(200, expectedresponse);
+    .reply(200, [{
+      access_token: expiredToken,
+      expiration_date: expirationDate,
+    }]);
 
-  const token = await edlObj.getEDLToken();
-  t.is(token, 'ABCDE');
+  const result = await retrieveEDLToken(username, password, 'SIT');
+
+  t.is(result, undefined);
 });
 
-test.serial('revokeToken revokes the user token', async (t) => {
-  const edlObj = new EarthdataLogin({
-    username: 'username',
-    password: 'password',
-    edlEnv: 'PROD',
+test('retrieveEDLToken returns the token if it expires later the same day', async (t) => {
+  // There is a race condition in this test that could pop up if the test is run near midnight
+
+  const { username, password } = t.context;
+  const now = new Date();
+
+  const fiveSecondsFromNow = new Date(now.valueOf() + (300 * 1000));
+
+  const token = createToken({
+    expirationTime: fiveSecondsFromNow.valueOf() / 1000,
   });
 
+  nock('https://sit.urs.earthdata.nasa.gov')
+    .get('/api/users/tokens')
+    .reply(200, buildGetTokensResponse([token]));
+  const result = await retrieveEDLToken(username, password, 'sit');
+
+  t.is(result, token);
+});
+
+test('retrieveEDLToken returns the last-expiring token if there are multiple tokens', async (t) => {
```

Comment:
suggest to rewrite this test, as Marc said, 3 tokens with the last-expiring in the middle, and retrieveEDLToken should return that one.
You can create the tokens with map function to avoid duplicate code.
Avoid naming like july/june, it makes the test hard to follow.